redhat-cop/helm-charts
# introduce chart install testing with ct install

#### What is this PR About?
adds the framework for doing  run time testing of chart installations using `ct install` against a kind cluster.

This is fully flushed out for the `operators-installer` and enabled for `arogcd-operator`, `bootstrap-project`, `kopf`, and `network-policy` all of which worked without any further modificiation. to enable this for other charts work will need to be done to further configre the kind cluster to mimic openshift or to make the charts (where possible) none openshift specific.

additionally adding `${CHART_NAME}/ci/${TEST_NAME}-values.yaml` files will create non default tests for any chart once it has been removed from the `INSTALL_TEST_EXCLUDED_CHARTS` list in the install-test.yaml.

#### How do we test this?
the ci is the testing.

this change is based ontop of https://github.com/redhat-cop/helm-charts/pull/350 so https://github.com/redhat-cop/helm-charts/pull/350 should be reviewed and merged first. this was done so that this MR could run CI for https://github.com/redhat-cop/helm-charts/pull/350 without conflating these CI changes with https://github.com/redhat-cop/helm-charts/pull/350

cc: @redhat-cop/day-in-the-life
